elements/lisk-bft/test/unit/finality_manager.spec.ts

```diff
@@ -94,20 +95,39 @@ describe('finality_manager', () => {
 						new FinalityManager({
 							chain: chainStub,
 							finalizedHeight,
+							genesisHeight: 0,
 							threshold,
 						}),
 				).toThrow('Invalid number of validators for BFT property');
 			});
 
-			it('should initialize maxHeightPrevoted to the finalizedHeight', async () => {
-				const nonZeroFinalizedHeight = 10000000;
-				finalityManager = new FinalityManager({
-					chain: chainStub,
-					finalizedHeight: nonZeroFinalizedHeight,
-					threshold,
-				});
-				await expect(finalityManager.getMaxHeightPrevoted()).resolves.toEqual(
-					nonZeroFinalizedHeight,
+			it('should initialize maxHeightPrevoted to the the last maxHeightPrevoted', async () => {
+				const validatorLedger = {
+					validators: [],
+					ledger: [
+						{
+							height: 10,
+							prevotes: 67,
+							precommits: 0,
+						},
+						{
+							height: 11,
+							prevotes: 67,
+							precommits: 0,
+						},
+						{
+							height: 12,
+							prevotes: 66,
+							precommits: 0,
+						},
+					],
+				};
+				jest
+					.spyOn(chainStub.dataAccess, 'getConsensusState')
+					.mockResolvedValue(codec.encode(BFTVotingLedgerSchema, validatorLedger));
+				const lastMaxHeightPrevoted = 30;
```

Comment:
I don't complete understand these specs, as the ledger state only contains 3 blocks and the `lastMaxHeightPrevoted` is larger than the height of the current tip of the chain.

Reply:
main case i wanted to create here is that state does not have any prevote higher than threshold, and maxHeightPrevoted not in the list.
but i guess `lastMaxHeightPrevoted` should be lower than the height in the state to be realistic

Reply:
Updated test case height to be more realistic